lxd/storage/backend_lxd.go

```diff
@@ -2549,6 +2531,51 @@ func (b *lxdBackend) CreateInstanceFromConversion(inst instance.Instance, conn i
 
 	revert.Add(func() { _ = b.driver.DeleteVolume(volCopy.Volume, op) })
 
+	// At this point, the instance's volume is populated. If "virtio" option is enabled,
+	// inject the virtio drivers.
+	if slices.Contains(args.ConversionOptions, "virtio") {
+		b.logger.Debug("Inject virtio drivers started")
+		defer b.logger.Debug("Inject virtio drivers finished")
+
+		err = b.driver.MountVolume(vol, op)
+		if err != nil {
+			return err
+		}
+
+		defer func() { _, _ = b.driver.UnmountVolume(vol, true, op) }()
+
+		diskPath, err := b.driver.GetVolumeDiskPath(vol)
+		if err != nil {
+			return err
+		}
+
+		imgFormat, _, err := b.qemuImageInfo(diskPath)
+		if err != nil {
+			return err
+		}
+
+		// Ensure the disk path points to a disk in raw format.
+		if imgFormat != "raw" {
```

Comment:
why is this check needed?

Should we have not already converted the volume to raw by now?

Reply:
Correct. This should never occur. Will remove